nixos/modules/services/continuous-integration/buildbot/master.nix

```diff
@@ -222,7 +222,7 @@ in {
   };
 
   config = mkIf cfg.enable {
-    users.groups = optional (cfg.group == "buildbot") {
+    users.groups = optionalAttrs (cfg.group == "buildbot") {
```

Comment:
wouldn't it be better to always use `mkIf` for subtrees in the module system?

Reply:
They're almost used interchangeably if you search for `users.groups = optionalAttrs` or `users.groups = mkIf` that I lack a preference.

Reply:
Though combinations of the two with `//` is actually a bug, ref https://github.com/NixOS/nixpkgs/pull/72949 and what seemed like a seemingly mysterious issue https://github.com/NixOS/nixpkgs/pull/72916. One should then be mindful to use `mkMerge` and `mkIf` only.